test/src/test/java/org/corfudb/infrastructure/logreplication/LogEntryReaderTest.java

```diff
@@ -0,0 +1,175 @@
+        int maxMsgSize = 300;
+        Set<Integer> indexAboveLimit = new HashSet<>();
+        indexAboveLimit.add(5);
+        verifyAllMessagesSent(maxMsgSize, indexAboveLimit);
+    }
+
+    private void verifyAllMessagesSent(int maxMsgSize, Set<Integer> indexesAboveLimit) {
+        createReplicationConfig(maxMsgSize);
+
+        // Generate deltas such that indexes in indexAboveLimit are bigger than maxMsgSize.
+        List<Long> seqNumbersWritten = generateDeltas(indexesAboveLimit);
+
+        // Start reading the deltas from LogEntryReader and collect the messages constructed
+        UUID requestId = UUID.randomUUID();
+        LogReplication.LogReplicationEntryMsg msgCreated;
+        List<LogReplication.LogReplicationEntryMsg> msgsSent = new ArrayList<>();
+
+        do {
+            msgCreated = logEntryReader.read(requestId);
+            if (msgCreated != null) {
+                msgsSent.add(msgCreated);
+            }
+        } while (msgCreated != null);
+
+        // Collect the sequence numbers of all constructed messages
+        List<Long> seqNumbersSent = new ArrayList<>();
+        for (LogReplication.LogReplicationEntryMsg msg : msgsSent) {
+            for (OpaqueEntry opaqueEntry : CorfuProtocolLogReplication.extractOpaqueEntries(msg)) {
+                seqNumbersSent.add(opaqueEntry.getVersion());
+            }
+        }
+        // Verify that all addresses which were written are included in the LogEntryMsgs constructed by LogEntryReader.
+        Assert.assertEquals(seqNumbersWritten, seqNumbersSent);
+    }
+
+    private void createReplicationConfig(int maxMsgSize) {
+        CorfuRuntime corfuRuntime = getNewRuntime(getDefaultNode()).connect();
+        LogReplicationConfigManager configManager = new LogReplicationConfigManager(corfuRuntime);
+        LogReplicationConfig config = new LogReplicationConfig(configManager,
+            LogReplicationConfig.DEFAULT_MAX_NUM_MSG_PER_BATCH,
+            maxMsgSize, LogReplicationConfig.MAX_CACHE_NUM_ENTRIES,
+            LogReplicationConfig.DEFAULT_MAX_SNAPSHOT_ENTRIES_APPLIED);
+        config.syncWithRegistry();
+        logEntryReader = new StreamsLogEntryReader(corfuRuntime, config);
+    }
+
+    private List<Long> generateDeltas(Set<Integer> indexesAboveLimit) {
+        // Create deltas by writing to testTable1.  If a delta must be larger than maxMsgSize, write to testTable2
+        // in addition to testTable1.
+        List<Long> seqNums = new ArrayList<>();
+
+        for (int i = 0; i < 10; i++) {
+            try (TxnContext txn = corfuStore.txn(namespace)) {
+                Sample.StringKey key = Sample.StringKey.newBuilder().setKey("key_" + i).build();
+                SampleSchema.ValueFieldTagOne val =
+                    SampleSchema.ValueFieldTagOne.newBuilder().setPayload("val_" + i).build();
+                txn.putRecord(table1, key, val, null);
+
+                if (indexesAboveLimit.contains(i)) {
```

Comment:
Wondering if it is possible to test using the actual transaction sizes the same way the LogEntryReader is doing (`ReaderUtility.calculateOpaqueEntrySize`) ?

Reply:
Yes it directly invokes the methods the LogEntryReader.  There is no customization in size calculation.